proto/agent.proto

```diff
@@ -325,10 +333,19 @@ message Connection {
 	uint64 totalBytesReceived = 14;
 }
 
+// ContainerMetadata only holds enough information to identify a container in connection data
+message ContainerMetadata {
+	string id = 1;
+	uint32 key = 2;
+	bytes byteKey = 3;
+	repeated string tags = 4;
+}
+
 message Addr {
-	Host host = 1;
+	Host host = 1;  // post-resolution field
 	string ip = 2;
 	int32 port = 3;
+	ContainerMetadata container = 4; // post-resolution field
```

Comment:
I'm a little worried about this adding this into Addr (as well as the pre-existing `Host`).

Since these objects contain string array tags, there's a high likelihood for duplication, which we've seen cause decoding performance issues for Processes as the message sizes increase quite a bit.

What do you think of changing this `Addr` object to store the `host_id` and/or `container_id`, and then top-level in `CollectorConnections`, we store two maps, `map<host_id, Host>` and `map<container_id, ContainerMetadata>`?

Reply:
I think that makes sense.  Removing the host field from addr is going to break some downstream code (none of which is critical, but it's still some code we'll need to rewrite)

Reply:
Done, I made this field deprecated. It can later become reserved.